### PR TITLE
feat(a11y): pagination aria

### PR DESCRIPTION
**Pull request by zwlafk**
[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [ ] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update
 - [ ] Refactor
 - [ ] Test Case
 - [ ] TypeScript definition update
 - [ ] Document improve
 - [ ] CI/CD improve
 - [ ] Branch sync
 - [ ] Other, please describe:

### PR description

Fixes #

### Changelog
🇨🇳 Chinese
- 修复 ...

---

🇺🇸 English
- Fix ...

### Checklist
- [ ] Test or no need
- [ ] Document or no need
- [ ] Changelog or no need

### CODE REVIEW

File: packages/semi-ui/pagination/index.tsx [hunk 1]
```diff
@@ -315,7 +315,7 @@ export default class Pagination extends BaseComponent<PaginationProps, Paginatio
                     onClick={() => this.foundation.goPage(page, i)}
                     className={pageListClassName}
                     tab-index={0}
-
+                    aria-label={page === '...' ? 'More' : `${page}`}
```
Comment (shijiatongxue) [inline packages/semi-ui/pagination/index.tsx:318]: 改成 `page ${page}` ？

File: packages/semi-ui/pagination/index.tsx [hunk 2]
```diff
@@ -223,7 +223,7 @@ export default class Pagination extends BaseComponent<PaginationProps, Paginatio
             [`${prefixCls}-next`]: true,
         });
         return (
-            <li onClick={e => !nextDisabled && this.foundation.goNext(e)} className={nextClassName} tab-index={0}>
+            <li aria-label="Next" onClick={e => !nextDisabled && this.foundation.goNext(e)} className={nextClassName} tab-index={0}>
```
Comment (shijiatongxue) [inline packages/semi-ui/pagination/index.tsx:226]: 如果把 label 放在 li 上，aria-diabled 已应该加上？然后把子级 aria-hidden 掉

Comment (shijiatongxue) [reply packages/semi-ui/pagination/index.tsx:226]: li 的 role 应该是 button？

File: packages/semi-ui/pagination/index.tsx [hunk 3]
```diff
@@ -208,7 +208,7 @@ export default class Pagination extends BaseComponent<PaginationProps, Paginatio
             [`${prefixCls}-item-disabled`]: prevDisabled,
         });
         return (
-            <li onClick={e => !prevDisabled && this.foundation.goPrev(e)} className={preClassName} tab-index={0}>
+            <li aria-label="Previous" onClick={e => !prevDisabled && this.foundation.goPrev(e)} className={preClassName} tab-index={0}>
```
Comment (shijiatongxue) [inline packages/semi-ui/pagination/index.tsx:211]: 改成 `Go to previous page`？

File: packages/semi-ui/pagination/index.tsx [hunk 4]
```diff
@@ -354,6 +354,7 @@ export default class Pagination extends BaseComponent<PaginationProps, Paginatio
                     className={className}
```
Comment (shijiatongxue) [inline packages/semi-ui/pagination/index.tsx:354]: 如果是当前页的话，加个 `aria-current=true`？

File: packages/semi-ui/pagination/index.tsx (hunk 1 above)
Comment (shijiatongxue) [reply packages/semi-ui/pagination/index.tsx:318]: image

这里加个 aria-label ？

**Comment by shijiatongxue** [comment]
可以补充下 Markdown

image

File: packages/semi-ui/pagination/index.tsx (hunk 3 above)
Comment (zwlafk) [reply packages/semi-ui/pagination/index.tsx:211]: > 改成 `Go to previous page`？

这种内置的 `aria-label` 是不是应该做成多语言写在locale里

File: packages/semi-ui/pagination/index.tsx [hunk 5]
```diff
@@ -213,7 +213,7 @@ export default class Pagination extends BaseComponent<PaginationProps, Paginatio
             [`${prefixCls}-item-disabled`]: prevDisabled,
         });
         return (
-            <li aria-label="Previous" onClick={e => !prevDisabled && this.foundation.goPrev(e)} className={preClassName} tab-index={0}>
+            <li role="button" aria-disabled={prevDisabled ? true : false} aria-label="Previous" onClick={e => !prevDisabled && this.foundation.goPrev(e)} className={preClassName} tab-index={0}>
```
Comment (shijiatongxue) [inline packages/semi-ui/pagination/index.tsx:216]:
```suggestion
            <li role="button" aria-disabled={prevDisabled ? true : false} aria-label="Previous" onClick={e => !prevDisabled && this.foundation.goPrev(e)} className={preClassName} tabindex={0}>
```

File: packages/semi-ui/pagination/index.tsx [hunk 6]
```diff
@@ -228,7 +228,7 @@ export default class Pagination extends BaseComponent<PaginationProps, Paginatio
             [`${prefixCls}-next`]: true,
         });
         return (
-            <li aria-label="Next" onClick={e => !nextDisabled && this.foundation.goNext(e)} className={nextClassName} tab-index={0}>
+            <li role="button" aria-disabled={nextDisabled ? true : false} aria-label="Next" onClick={e => !nextDisabled && this.foundation.goNext(e)} className={nextClassName} tab-index={0}>
```
Comment (shijiatongxue) [inline packages/semi-ui/pagination/index.tsx:231]:
```suggestion
            <li role="button" aria-disabled={nextDisabled ? true : false} aria-label="Next" onClick={e => !nextDisabled && this.foundation.goNext(e)} className={nextClassName} tabindex={0}>
```

**Pull request merged**